pkg/proxy/proxy.go

```diff
@@ -185,32 +205,91 @@ func (p *Proxy) forwardRequest(host string, w http.ResponseWriter, req *http.Req
 		// If the client sent multiple XFF headers, flatten them.
 		proxyReq.Header.Set(xff, strings.Join(previous, ", "))
 	}
-	proxyReq.URL.Host = host
 	proxyReq.URL.Scheme = "https"
+	attempts := 0
 
-	log.Debug("Forwarding request to %s", proxyReq.URL.String())
-	client := http.Client{}
-	resp, err := client.Do(proxyReq)
-	if err != nil {
-		if urlErr, ok := err.(*url.Error); ok && urlErr.Timeout() {
-			writeJSONError(w, http.StatusGatewayTimeout, urlErr)
-		} else {
+	var requestBody []byte
+	if req.Body != nil {
+		requestBody, err = io.ReadAll(req.Body)
+		if err != nil {
 			writeJSONError(w, http.StatusBadGateway, err)
+			return
 		}
-		return
+		req.Body = io.NopCloser(bytes.NewBuffer(requestBody))
 	}
-	defer resp.Body.Close()
 
-	for _, hdr := range connectionHeaders {
-		resp.Header.Del(hdr)
-	}
-	outHeader := w.Header()
-	for name, value := range resp.Header {
-		outHeader[name] = value
-	}
+	for {
+		proxyReq.URL.Host = acct.Host
+		log.Debug("Forwarding request to %s", proxyReq.URL.String())
+		client := http.Client{}
+		result, err := client.Do(proxyReq)
```

Comment:
Sorry to resurrect this but I worry this isn't going to work. This is assigned a request body `req.Body` at the start of the function on line 182, but on line 213 we drain `req.Body` completely and I worry this thing will never have a body.

In testing, stuff like [fleet_status](https://developer.tesla.com/docs/fleet-api/endpoints/vehicle-endpoints#fleet-status) is failing every time when Tesla validates the request body.

Reply:
Maybe on line 218 we should write
```
		proxyReq.Body = io.NopCloser(bytes.NewBuffer(requestBody))
```
?

Reply:
Thanks for testing this out. Indeed. the proxyReq is the real request being sent out. Will fxi it.

Reply:
https://github.com/teslamotors/vehicle-command/pull/327